metricbeat/mb/module/wrapper_test.go

```diff
@@ -193,14 +193,20 @@ func TestPeriodIsAddedToEvent(t *testing.T) {
 			require.NoError(t, err)
 
 			done := make(chan struct{})
-			defer close(done)
 
 			output := m.Start(done)
 
 			event := <-output
 
 			hasPeriod, _ := event.Fields.HasKey("metricset.period")
 			assert.Equal(t, c.hasPeriod, hasPeriod, "has metricset.period in event %+v", event)
+
+			// stop worker
+			close(done)
+
+			// wait for shutdown
+			for range output {
+			}
```

Comment:
(nit) even more accurate:
```suggestion
			// wait for shutdown
			event, ok := <-output
			assert.Falsef(t, ok, "received unexpected event: %+v", event)
```